Automation/NSG/deploy-nsg.ps1

```diff
@@ -0,0 +1,97 @@
+param(
+    # Change value if you want a different region limit notification
+    $ResourceLimit = 15,
+    $TemplateFilePath = "./template.json",
+    $ParameterFilePath = "./parameters.json",
+
+    # Found in Azure secrets
+    [Parameter(Mandatory=$True)]
+    [string]
+    $ServicePrincipalUsername,
+
+    # Your tenant Id in Azure
+    [Parameter(Mandatory=$True)]
+    [string]
+    $TenantId,
+
+    # Name for the resource group you will create
+    [Parameter(Mandatory=$True)]
+    [string]
+    $ResourceGroupName,
+
+    # Region. ex. westus, eastus...
+    [Parameter(Mandatory=$True)]
+    [string]
+    $Region,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $DeploymentName
+)
+
+# Service principal login, will prompt user for password
+Write-Host "`n"
+$credential = Get-Credential -Username $ServicePrincipalUsername
+
+Connect-AzAccount `
```

Comment:
Why don't you grab my login.ps1 file and put it in your Automation folder and then you can just call it from this script? For some reason I'm getting errors from this part of the script:

`Connect-AzAccount: D:\VSCode\ad440-winter2021-thursday-repo\Automation\NSG\deploy-nsg.ps1:36:1
Line |
  36 |  Connect-AzAccount `
     |  ~~~~~~~~~~~~~~~~~~~
     | ClientSecretCredential authentication failed: A configuration issue is preventing authentication - check
     | the error message from the server for details.You can modify the configuration in the application
     | registration portal. See https://aka.ms/msal-net-invalid-client for details.  Original exception:
     | AADSTS7000215: Invalid client secret is provided. Trace ID: c2f72355-bb23-4cb7-aae2-488b4e7bce00
     | Correlation ID: 237a5131-c24b-4c0d-8235-2a97e235018f Timestamp: 2021-01-21 06:19:47Z`

Reply:
Oh I must have misunderstood your initial comment about not being able to log in! Yes Afrasiyab was having the same error with his script, I will check with him on a solution.

Reply:
The login.ps1 is working well - if my PR can get approved/merged, other people (Afrasiyab) can use that file